### PR TITLE
Viper integrate compiler

### PR DESCRIPTION
**Pull request by jacqueswww**
### What was wrong?
Viper wasn't supported :tongue: 

### How was it fixed?
To test you can simply install viper (`pip install https://github.com/ethereum/viper/archive/master.zip`) (https://github.com/ethereum/populus/pull/383/ should be merged by the time of review ;)). Viper requires python3.6 virtualenv.

Added the necessary code in ViperBackend. Currently you have to specify the backend as part of your project.json as follows:

```json
{
    "version": "7",
    "compilation": {
        "contracts_source_dirs": ["./contracts"],
        "import_remappings": [],
        "backend": {
            "class": "populus.compilation.backends.ViperBackend"
        }
    }
}
```

```json
{
    "version": "7",
    "compilation": {
        "contracts_source_dirs": ["./contracts"],
        "import_remappings": [],
        "backend": {
            "class": "populus.compilation.backends.ViperBackend"
        }
    }
}
```

Then you can add .v.py file in contracts path as follows (this replicates default init Greeter contract):
```python
$ cat contracts/Greeter.v.py 
greeting: bytes <= 20

@public
def __init__():
    self.greeting = "Hello"

@public
def setGreeting(x: bytes <= 20):
    self.greeting = x

@public
def greet() -> bytes <= 40:
    return self.greeting

```

And then run py.test, and Greeter.sol has just been replicated!

### CODE REVIEW

**Comment by jacqueswww** [comment]
I added a very basic  test for testing the backend if viper is installed. I think at this moment in time we shouldn't worry too much about placing viper in setup.py ?

**Comment by jacqueswww** [comment]
I managed to get the tests running under their own py3.6 build ;)

File: populus/compilation/backends/viper.py [hunk 1]
```diff
@@ -1,8 +1,38 @@
...
-    project_source_extensions = ('*.v.py', '*.vy')
-    test_source_extensions = ('*.py', )
+    project_source_glob = ('*.v.py', '*.vy')
+    test_source_glob = ('test_*.v.py', 'test_*.vy')
+
+    def get_compiled_contracts(self, source_file_paths, import_remappings):
+        try:
+            from viper import compiler
+        except ImportError:
+            raise Exception('viper needs to be installed to use ViperBackend as compiler backend.')
```
Comment (pipermerriam) [inline populus/compilation/backends/viper.py]: I think using `ImportError` as the exception class here would be better than the bare base `Exception` class.

File: populus/config/versions.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 V5 = '5'
 V6 = '6'
 V7 = '7'
+V7_VIPER = '7-viper'
```
Comment (pipermerriam) [inline populus/config/versions.py]: This appears to be un-used?

Comment (jacqueswww) [reply populus/config/versions.py]: Ha yes! Must've slipped in from when I was still figuring out things ;)

**Comment by jacqueswww** [comment]
@pipermerriam thanks for the review. :smile_cat:  I have added the preliminary docs, please have a look and add the necessary warnings (viper being experimental) - I wasn't exactly sure where to put it?

File: docs/index.rst [hunk 3]
```diff
@@ -24,6 +24,7 @@ Contents
     resources
     modules
     release
+    viper_support
```
Comment (pipermerriam) [inline docs/index.rst]: I don't see a corresponding `docs/viper_support.rst` file.  Did it not get committed.

Comment (jacqueswww) [reply docs/index.rst]: Woops, fixing!

Comment (jacqueswww) [reply docs/index.rst]: @pipermerriam added the file now :) sorrry again!

Comment (veox) [reply docs/index.rst]: @jacqueswww I still can't see it. :(

Comment (jacqueswww) [reply docs/index.rst]: @veox try now, must've been a force push fail :(

File: docs/viper_support.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
+Viper Support
+=============
```
Comment (jacqueswww) [inline docs/viper_support.rst]: @veox finally! hehehe

**Comment by pipermerriam** [comment]
restarting the tests that failed.  This looks good.  Thanks for contributing.  I'll get it merged and released hopefully today.

**Comment by jacqueswww** [comment]
@pipermerriam awesome, thanks! :smile: :man_dancing:

**Pull request merged**